docs/perf/nsight_profiling.rst

```diff
@@ -0,0 +1,122 @@
+Nsight Profiling in verl
+==============================
+
+This guide explains how to use NVIDIA Nsight Systems for profiling verl training runs.
+
+Configuration
+--------------
+
+Profiling in verl can be configured through several parameters in the trainer configuration file (ppo_trainer.yaml or other files like dapo_trainer.yaml):
+
+0. Prerequisites
+
+Nsight Systems version is important, please reference ``docker/Dockerfile.vllm.sglang.megatron`` for the version we used.
+
+1. Global Profiling Control
+--------------------------
+
+``trainer.profile_steps``
+    List of step numbers at which profiling should be performed. For example: [1, 2, 5] will profile steps 1, 2, and 5. And ``null`` means no profiling.
+
+verl has one single controller process and multiple worker processes. Both controller and worker processes can be profiled. Since the controller process can be executed in any nodes in the cluster, there is message printed in the logging to indicate the controller process node hostname and process id.
+
+2. Profiling rank control
+----------------
+
+By default, verl profiles the whole training process in a single `` worker_process_<PID>.<step>.nsys-rep`` file for each process rank. If you want to profile only part of the process ranks, you can set the ``profile_ranks`` parameter for each component. ``null`` and ``[]`` means no ranks, and ``[0,1]`` means profile ranks 0 and 1. If you want to profile all ranks, you can set the ``profile_ranks_all`` parameter for each component.
+
```

Comment:
FYI the doc website supports markdown as well. I think displaying them with bullet points makes more sense? 
![image](https://github.com/user-attachments/assets/3514cf0b-1b68-4918-8136-a17426daa014)

Reply:
I also update the PR description.